tests/components/modbus/test_modbus_sensor.py

```diff
@@ -546,6 +547,16 @@ async def test_all_sensor(hass, cfg, regs, expected):
             [0x0101],
             "257",
         ),
+        (
+            {
+                CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 4,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">HH",
+            },
+            [0x1122, 0x3344],
```

Comment:
I can't use 0x11223344?

Reply:
That depend on register_size and what you want returned. As it is it returns 2 registers each 2 bytes.

Reply:
Count 1 byte 4

I think that is tests engine error

Because work in real world

Only fail on tests....

Reply:
Yes the config with 2 registers will produce an error if you specify count: 1, that is logical.

But don't only think, debug and find the problem.

Reply:
This test pass as it is. Soo...

Reply:
![20210509_230128](https://user-images.githubusercontent.com/45373783/117589119-92cdde80-b11f-11eb-840e-7621891372e2.jpg)
If I use count 1 size 4 --- [0x11223344]